### DISCUSSION TITLE
Method cell[i].points vs. deprecated method cell_points(i), New method is tremendously slower

### DISCUSSION THREAD

**Discussion by LukeDiederich** [pyvista/pyvista#4314] [Q&A] [answered]
Hi,

a part of my code relies on iterating over cells and getting the points of the equivalent cell. 
After updating to Python 3.11 and having a newer version of PyVista as well I got the following message: 

PyVistaDeprecationWarning: `cell_points` is deprecated. Use `cell[i].points` instead

However for me it seems that the newer method (cell[i].points) is tremendously slower, which slows down my code by a large margin.

Here is an example code to test this:
```
import pyvista as pv
import time

mesh = pv.Sphere()
mesh = mesh.subdivide(1, 'loop')
print(mesh.n_cells)
timer = time.time()
list_points = []
all_cells = list(range(0, mesh.n_cells))
for i in all_cells:
    points = mesh.cell[i].points
    list_points.append(points)

print("--- %s seconds for new method ---" % (time.time() - timer))

timer2 = time.time()
list_points = []
all_cells = list(range(0, mesh.n_cells))
for i in all_cells:
    points = mesh.cell_points(i)
    list_points.append(points)

print("--- %s seconds for old method ---" % (time.time() - timer2))
```

Time for new method: --- 813.1108591556549 seconds for new method ---
Time for old method: --- 0.16899967193603516 seconds for old method ---

Are there any plans in the future to remove the method of cell_points(i)? Or can it be used in the future without any worries?
Or am I using the method wrong? I am wondering why it is that much slower...

Thanks for any help

Best regards 
Luke

**Comment by MatthewFlamm** [comment, marked as answer]
The suggested use of `cell[i]` was a mistake.  This has been fixed in #4082, which will be in the next release.

You should use `get_cell(i)` instead.

**Reply by LukeDiederich** [reply]
Thanks! Works fine and is faster than the previous method as well:

   ` points = mesh.get_cell(i).points`
